test/unit/watcher.c

```diff
@@ -519,12 +521,19 @@ main_f(va_list ap)
 int
 main()
 {
+	struct lua_State *L = luaL_newstate();
+	luaL_openlibs(L);
+	tarantool_L = L;
+
```

Comment:
Creating a Lua state for a test that doesn't use Lua looks ugly. I see a few alternatives:

1. We could check `tarantool_L` in the boxlib implementation of `cord_on_yield`. Then we could drop `core_test_utils.c` altogether. One `if` should be negligible performance-wise. I like this option most.
2. It's also possible to call `cord_on_yield` by a function pointer and overwrite the pointer explicitly in tests.
3. To avoid a callback, we could mark the `cord_on_yield` definition as weak so that the linker would always select the definition from `core_test_utils.c`.
4. Instead of using weak references, which are platform-dependent, we could use macros (e.g. wrap boxlib `cord_on_yield` in `#ifndef TT_UNIT_TEST` macro).

WDYT?

Reply:
I would just keep this patch as is. Because libbox uses Lua anyway. For instance, you can't call `box_cfg()` or `box_set_replication()` without Lua being initialized. That looks like a bigger issue.

Options (1) and (2) were tried AFAIK and showed worse performance. Especially notable was (2). @igormunkin can tell more.

(3) is platform-specific, isn't it? Will it work on Mac?

(4) looks uglier IMHO, but I don't mind if it would be done. Could be useful for other stuff too, for compile-time error injections.

Another option would be to make `box_init()` do this Lua initialization in case it sees it wasn't initialized yet.

Also `cord_on_yield` could be moved out of libbox into some other library used in the final executable, but not in any of unit tests. Maybe just build it together with `main.cc`.

Reply:
> Because libbox uses Lua anyway. For instance, you can't call box_cfg() or box_set_replication() without Lua being initialized.

Yes, but there are parts of boxlib, like watchers, that don't need this.

> Options (1) and (2) were tried AFAIK and showed worse performance.

I doubt that one `if` could affect performance anyhow.

> I would just keep this patch as is.

Okay.

Reply:
> > Options (1) and (2) were tried AFAIK and showed worse performance.
>
> I doubt that one `if` could affect performance anyhow.

Here is [the patchset](https://lists.tarantool.org/tarantool-patches/cover.1600862684.git.imun@tarantool.org/) mentioned by @Gerold103. You can also see the [v1 thread](https://lists.tarantool.org/tarantool-patches/b77be75a94623c0b430875ec1ad292579f86a613.1585606573.git.imun@tarantool.org/) for more info.